data_model/src/block.rs

```diff
@@ -0,0 +1,730 @@
+//! This module contains `Block` structures for each state, it's
+//! transitions, implementations and related traits
+//! implementations. `Block`s are organised into a linear sequence
+//! over time (also known as the block chain).  A Block's life-cycle
+//! starts from `PendingBlock`.
+
+#[cfg(not(feature = "std"))]
+use alloc::{boxed::Box, format, string::String, vec::Vec};
+use core::{cmp::Ordering, fmt::Display, iter};
+
+use derive_more::Display;
+use iroha_crypto::{HashOf, MerkleTree, SignatureOf, SignaturesOf};
+use iroha_schema::IntoSchema;
+use iroha_version::{declare_versioned_with_scale, version_with_scale};
+use parity_scale_codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+
+pub use self::{
+    candidate::{CandidateBlock, VersionedCandidateBlock},
+    candidate_committed::{CandidateCommittedBlock, VersionedCandidateCommittedBlock},
+    chained::ChainedBlock,
+    committed::{CommittedBlock, VersionedCommittedBlock},
+    header::BlockHeader,
+    pending::{PendingBlock, VersionedPendingBlock},
+    signed::SignedBlock,
+    valid::ValidBlock,
+};
```

Comment:
IMO the only block that we need in `data_model` is `CommittedBlock`

Reply:
we don't want internal state transitions polluting our public API

Reply:
i agree